samcli/lib/providers/provider.py

```diff
@@ -171,6 +178,11 @@ def _compute_layer_name(is_defined_within_template, arn):
             [layer_name, layer_version, hashlib.sha256(arn.encode("utf-8")).hexdigest()[0:10]]
         )
 
+    @staticmethod
+    def _verify_code_uri_existence(code_uri):
+        if not os.path.exists(code_uri):
+            raise InvalidLayerVersionContentUri(code_uri + " is an invalid ContentUri, file/directory does not exist.")
```

Comment:
What if the Layer is defined to be an S3 URI already?

Reply:
good catch!

Reply:
It does not seem to affect it. I tested on this template snippet:

```yarml
Resources:
  TestLambdaLayer:
    Type: AWS::Serverless::LayerVersion
    Properties:
      LayerName: test-layer
      Description: Lambda layer with all node module dependency and shared code between
        lambda
      ContentUri: s3://dummy-bucket/my-layer.zip
      CompatibleRuntimes:
      - nodejs12.x
      RetentionPolicy: Retain
```

The `code_url`'s value is `.`, so it will pass the `_verify_code_uri_existence`